packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1928,6 +1933,17 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
         context: ErrorDescription('while calling onSubmitted for $action'),
       ));
     }
+
+    // Create a new TextInputConnection to replace the current one. This
+    // on iOS switches to a new input view and on Android restarts the
+    // input method. In both cases this call resets the IME.
+    //
+    // We do this for fidelity reasons, in case the developer calls
+    // requestFocus in the onSubmitted callback to undo the focus change.
+    // See https://github.com/flutter/flutter/issues/84240.
+    if (shouldUnfocus) {
+      _restartConnectionIfNeeded();
```

Comment:
is this the only use case where we want to restart the connection? what about programmatically changing the focus?

Reply:
Will _finalizeEditing still be called in that case?  It looks like anything that closes the connection will call it.  I'm also curious if there are any other cases that this PR wouldn't cover, can't think of any others though.

Reply:
Yes currently that's the only case we the we want to restart the connection to restart the IME from the framework.